src/array/schema.rs

```diff
@@ -248,4 +301,40 @@ mod tests {
             assert!(s.allows_duplicates());
         }
     }
+
+    #[test]
+    fn test_load() -> io::Result<()> {
+        let tmp_dir = TempDir::new("tiledb_array_schema_test_load")?;
+
+        let c: Context = Context::new().unwrap();
+
+        let r = create_quickstart_dense(&tmp_dir, &c);
+        assert!(r.is_ok());
+
+        let schema = Schema::open(&c, &r.unwrap())
+            .expect("Could not open quickstart_dense schema");
+
+        let domain = schema.domain();
+
+        let rows = domain.dimension(0).expect("Error reading rows dimension");
+        assert_eq!(Datatype::Int32, rows.datatype());
```

Comment:
Just to illustrate the lifetimes, if you were to put this in a loop so that you called this N times to create the dimension and then drop it, internally it'd be just fine. Shoving the returned dimensions into a vector or similar would also be fine. You'd have N distinct allocated handle objects that all proxy through to the underlying internal `shared_ptr<tiledb::sm::Dimension>`.